### PR TITLE
fix(auth): replace SecureCredentialStore with native OS credential storage

### PR DESCRIPTION
**Pull request by joshsmithxrm**
## Summary

- **Fixes credential persistence bug** - The original `SecureCredentialStore` misused `MsalCacheHelper`, only checking if encryption was available but then writing plaintext JSON via `File.WriteAllBytesAsync()`, bypassing MSAL entirely. This caused ClientSecret, CertificateFile, and UsernamePassword auth to fail after token expiration.
- **Adds native OS credential storage** via `Devlooped.CredentialManager`:
  - Windows: Credential Manager (DPAPI encryption)
  - macOS: Keychain Services  
  - Linux: libsecret (with `--accept-cleartext-caching` fallback for CI/CD)
- **Automatic legacy file migration** - Existing plaintext credentials are migrated to native storage on first read
- **Adds ADR-0032** documenting the design decision

## Test plan

- [x] Unit tests pass with mocked `ICredentialStore`
- [x] Build passes on all target frameworks (net8.0, net9.0, net10.0)
- [ ] Manual: Create profile with ClientSecret, let token expire, verify re-auth works
- [ ] Manual: Verify credentials persist across CLI restarts

Partially addresses #133 (adds unit tests for credential store)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/PPDS.Auth/Credentials/NativeCredentialStore.cs [hunk 1]
```diff
@@ -0,0 +1,445 @@
...
+    /// </summary>
+    private async Task MigrateLegacyIfNeededAsync()
+    {
+        if (_migrationAttempted)
+            return;
+
+        await _migrationLock.WaitAsync().ConfigureAwait(false);
+        try
+        {
+            if (_migrationAttempted)
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply src/PPDS.Auth/Credentials/NativeCredentialStore.cs]: Fixed in f819199. Added pragma to suppress this false positive - the analyzer doesn't understand the double-checked locking pattern. The first check (line 205) is outside the lock, and a concurrent thread may have set the flag while this thread was waiting for the lock at line 208.

File: docs/internal/CREDENTIAL_STORE_FIX.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
+# Fix: Credential Store Persistence Bug
+
+**Status:** Implementation Plan
+**Branch:** `fix/credential-store-persistence`
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply docs/internal/CREDENTIAL_STORE_FIX.md]: Fixed in f819199. Updated status to 'Completed (superseded by ADR-0032)'.

File: docs/adr/0032_NATIVE_CREDENTIAL_STORE.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    _store = store ?? CredentialManager.Create(ServiceName);
+}
+```
+
+## References
+
+- [Devlooped.CredentialManager](https://github.com/devlooped/CredentialManager)
+- [Git Credential Manager Credential Stores](https://github.com/git-ecosystem/git-credential-manager/blob/main/docs/credstores.md)
+- ADR-0024: Shared Local State Architecture
+- `docs/internal/CREDENTIAL_STORE_FIX - Option B.md` - Original implementation plan
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply docs/adr/0032_NATIVE_CREDENTIAL_STORE.md]: Fixed in f819199. Corrected reference to docs/internal/CREDENTIAL_STORE_FIX.md.

File: src/PPDS.Cli/Commands/Auth/AuthCommandGroup.cs [hunk 4]
```diff
@@ -214,7 +214,7 @@ private static async Task<int> ExecuteCreateAsync(CreateOptions options, Cancell
 
         // Store secrets in secure credential store (not in profile), unless bypassed
         // Declared outside try block so finally can dispose it
-        SecureCredentialStore? credentialStore = null;
+        NativeCredentialStore? credentialStore = null;
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply src/PPDS.Cli/Commands/Auth/AuthCommandGroup.cs:217]: False positive. The credentialStore is conditionally created at line 252 and used in the CleanupCredentialsAsync local function (lines 289-295) which is called in multiple error branches. A using block would incorrectly scope its lifetime. The manual dispose pattern is correct here.

File: src/PPDS.Cli/Services/Profile/ProfileService.cs [hunk 5]
```diff
@@ -242,7 +242,7 @@ public async Task<ProfileSummary> CreateProfileAsync(
         var envSpnSecret = System.Environment.GetEnvironmentVariable(CredentialProviderFactory.SpnSecretEnvVar);
         var bypassCredentialStore = !string.IsNullOrWhiteSpace(envSpnSecret);
 
-        SecureCredentialStore? credentialStore = null;
+        NativeCredentialStore? credentialStore = null;
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply src/PPDS.Cli/Services/Profile/ProfileService.cs:245]: False positive. Same pattern as AuthCommandGroup - the credentialStore is conditionally created and needs to live across the entire method. A using block would incorrectly scope its lifetime.

File: tests/PPDS.Auth.Tests/Credentials/NativeCredentialStoreTests.cs [hunk 6]
```diff
@@ -0,0 +1,264 @@
...
+        Assert.NotNull(retrieved);
+        Assert.Equal("updated-secret", retrieved.ClientSecret);
+    }
+
+    [Fact]
+    public void IsCleartextCachingEnabled_WithoutFallback_ReturnsFalse()
+    {
+        // Assert
+        Assert.False(_store.IsCleartextCachingEnabled);
+    }
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply tests/PPDS.Auth.Tests/Credentials/NativeCredentialStoreTests.cs:198]: Good suggestion. Migration tests would be valuable but adding them in this PR risks scope creep. Filed as follow-up work item.

File: tests/PPDS.Auth.Tests/Credentials/NativeCredentialStoreTests.cs [hunk 7]
```diff
@@ -0,0 +1,264 @@
...
+
+        // Act
+        await _store.StoreAsync(credential);
+        var retrieved = await _store.GetAsync("test-app-id");
+
+        // Assert
+        Assert.NotNull(retrieved);
+        Assert.Equal("test-app-id", retrieved.ApplicationId);
+        Assert.Equal("test-secret", retrieved.ClientSecret);
+    }
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply tests/PPDS.Auth.Tests/Credentials/NativeCredentialStoreTests.cs:43]: Good suggestion. Adding validation edge case tests would be valuable but this PR focuses on the core credential persistence fix. Filed as follow-up.

File: src/PPDS.Auth/Credentials/NativeCredentialStore.cs [hunk 8]
```diff
@@ -0,0 +1,445 @@
...
+        foreach (var key in manifest)
+        {
+            _store.Remove(ServiceName, key);
+        }
+
+        // Clear the manifest itself
+        _store.Remove(ServiceName, ManifestKey);
+
+        return Task.CompletedTask;
+    }
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply src/PPDS.Auth/Credentials/NativeCredentialStore.cs:174]: The cancellationToken is required by the ISecureCredentialStore interface for API consistency across all implementations. Removing it would require changing the interface contract. The parameter is reserved for future use if we switch to async-native credential stores.

File: src/PPDS.Auth/Credentials/NativeCredentialStore.cs [hunk 9]
```diff
@@ -0,0 +1,445 @@
...
+                    _store.AddOrUpdate(ServiceName, key, kvp.Value);
+                    AddToManifest(key);
+                    migratedCount++;
+                }
+
+                // Securely delete the legacy file
+                SecureDeleteFile(_legacyFilePath);
+
+                // Log migration
+                Console.Error.WriteLine($"Migrated {migratedCount} credential(s) from legacy file to OS credential store.");
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply src/PPDS.Auth/Credentials/NativeCredentialStore.cs]: Migration happens once on first credential access - it's infrastructure that runs silently unless there's an error. The Console.Error messages only fire during migration (legacy file detected) and only for the count or if migration fails. Adding IProgressReporter for a one-time migration event that most users will never see would be over-engineering. The scope is limited to this single file.

File: src/PPDS.Auth/Credentials/NativeCredentialStore.cs [hunk 10]
```diff
@@ -0,0 +1,445 @@
...
+    /// <summary>
+    /// Configures the GCM credential store backend via environment variable.
+    /// </summary>
+    private static void ConfigureCredentialStoreBackend(bool allowCleartextFallback)
+    {
+        if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux) && allowCleartextFallback)
+        {
+            // CI/CD mode: use plaintext store when libsecret unavailable
+            // This matches PAC CLI behavior for headless environments
+            Environment.SetEnvironmentVariable("GCM_CREDENTIAL_STORE", "plaintext");
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply src/PPDS.Auth/Credentials/NativeCredentialStore.cs:102]: Acknowledged. The environment variable affects only this process and its children, and is set once per NativeCredentialStore instance. In practice, PPDS CLI is the root process and doesn't spawn child processes that use GCM. The GCM library doesn't expose per-instance configuration for the credential store backend - the environment variable is the documented way to configure it.

File: src/PPDS.Auth/Credentials/NativeCredentialStore.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply src/PPDS.Auth/Credentials/NativeCredentialStore.cs]: Fixed in f819199. Added pragma to suppress the false positive - this is a double-checked locking pattern for thread safety.

File: src/PPDS.Cli/Commands/Auth/AuthCommandGroup.cs [hunk 11]
```diff
@@ -214,7 +214,7 @@
...
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply src/PPDS.Cli/Commands/Auth/AuthCommandGroup.cs:217]: False positive. The credentialStore is conditionally created and used in a local function CleanupCredentialsAsync that's called in multiple error branches. A using block would incorrectly scope its lifetime.

File: src/PPDS.Cli/Services/Profile/ProfileService.cs [hunk 12]
```diff
@@ -242,7 +242,7 @@
...
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply src/PPDS.Cli/Services/Profile/ProfileService.cs:245]: False positive. Same pattern - the credentialStore is conditionally created and needs to live across the entire method for cleanup in error cases.

File: tests/PPDS.Auth.Tests/Credentials/NativeCredentialStoreTests.cs [hunk 13]
```diff
@@ -0,0 +1,264 @@
...
+
+        public IList<string> GetAccounts(string service)
+        {
+            var prefix = $"{service}|";
+            var accounts = new List<string>();
+            foreach (var key in _credentials.Keys)
+            {
+                if (key.StartsWith(prefix, StringComparison.OrdinalIgnoreCase))
+                {
+                    accounts.Add(key.Substring(prefix.Length));
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply tests/PPDS.Auth.Tests/Credentials/NativeCredentialStoreTests.cs:233]: Addressed - removed migration logic entirely. Pre-release software didn't need migration from broken storage.

File: src/PPDS.Auth/Credentials/NativeCredentialStore.cs [hunk 14]
```diff
@@ -0,0 +1,445 @@
...
+            }
+            catch (JsonException)
+            {
+                // Corrupted legacy file - just delete it
+                SecureDeleteFile(_legacyFilePath);
+            }
+            catch (Exception ex)
+            {
+                // Log but don't fail - user can still authenticate
+                Console.Error.WriteLine($"Warning: Failed to migrate legacy credentials: {ex.Message}");
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply src/PPDS.Auth/Credentials/NativeCredentialStore.cs]: Addressed - removed migration logic entirely. Console.Error output removed with it.

File: src/PPDS.Auth/Credentials/NativeCredentialStore.cs [hunk 15]
```diff
@@ -0,0 +1,445 @@
...
+        if (string.IsNullOrWhiteSpace(credential.ApplicationId))
+            throw new ArgumentException("ApplicationId is required.", nameof(credential));
+
+        var key = credential.ApplicationId.ToLowerInvariant();
+        var json = SerializeCredential(credential);
+
+        _store.AddOrUpdate(ServiceName, key, json);
+        AddToManifest(key);
+
+        return Task.CompletedTask;
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply src/PPDS.Auth/Credentials/NativeCredentialStore.cs:126]: Interface contract requirement for API consistency across implementations.

File: src/PPDS.Auth/Credentials/NativeCredentialStore.cs [hunk 16]
```diff
@@ -0,0 +1,445 @@
...
+        var key = applicationId.ToLowerInvariant();
+        var removed = _store.Remove(ServiceName, key);
+
+        if (removed)
+        {
+            RemoveFromManifest(key);
+        }
+
+        return Task.FromResult(removed);
+    }
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply src/PPDS.Auth/Credentials/NativeCredentialStore.cs:159]: Interface contract requirement for API consistency across implementations.

File: tests/PPDS.Auth.Tests/Credentials/NativeCredentialStoreTests.cs [hunk 17]
```diff
@@ -0,0 +1,264 @@
...
+                }
+            }
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply tests/PPDS.Auth.Tests/Credentials/NativeCredentialStoreTests.cs:235]: Addressed - removed migration logic entirely, test code simplified.
